fullstop-plugins/fullstop-application-lifecycle-plugin/src/main/java/org/zalando/stups/fullstop/plugin/LifecyclePlugin.java

```diff
@@ -0,0 +1,135 @@
+import org.zalando.stups.fullstop.violation.entity.ApplicationEntity;
+import org.zalando.stups.fullstop.violation.entity.LifecycleEntity;
+import org.zalando.stups.fullstop.violation.entity.VersionEntity;
+import org.zalando.stups.fullstop.violation.service.impl.ApplicationLifecycleServiceImpl;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.zalando.stups.fullstop.events.CloudtrailEventSupport.*;
+
+/**
+ * Created by gkneitschel.
+ */
+@Component
+public class LifecyclePlugin extends AbstractFullstopPlugin {
+
+    private static final Logger LOG = LoggerFactory.getLogger(LifecyclePlugin.class);
+
+    private static final String EC2_SOURCE_EVENTS = "ec2.amazonaws.com";
+
+    private static final String RUN_EVENT_NAME = "RunInstances";
+
+    private static final String START_EVENT_NAME = "StartInstances";
+
+    private static final String STOP_EVENT_NAME = "StopInstances";
+
+    private static final String TERMINATE_EVENT_NAME = "TerminateInstances";
+
+    private ApplicationLifecycleServiceImpl applicationLifecycleService;
+
+    private UserDataProvider userDataProvider;
+
+    @Autowired
+    public LifecyclePlugin(final ApplicationLifecycleServiceImpl applicationLifecycleService,
+            UserDataProvider userDataProvider) {
+        this.applicationLifecycleService = applicationLifecycleService;
+        this.userDataProvider = userDataProvider;
+    }
+
+    @Override
+    public boolean supports(CloudTrailEvent event) {
+        CloudTrailEventData cloudTrailEventData = event.getEventData();
+        String eventSource = cloudTrailEventData.getEventSource();
+        String eventName = cloudTrailEventData.getEventName();
+
+        return eventSource.equals(EC2_SOURCE_EVENTS) &&
+                (eventName.equals(RUN_EVENT_NAME) ||
+                        eventName.equals(START_EVENT_NAME) ||
+                        eventName.equals(STOP_EVENT_NAME) ||
+                        eventName.equals(TERMINATE_EVENT_NAME)
+                );
+    }
+
+    @Override
+    public void processEvent(CloudTrailEvent event) {
+        List<String> instances = getInstances(event);
+        for (String instance : instances) {
+            DateTime eventDate = getLifecycleDate(event, instance);
+
+            LifecycleEntity lifecycleEntity = new LifecycleEntity();
+            lifecycleEntity.setEventType(event.getEventData().getEventName());
+            lifecycleEntity.setEventDate(eventDate);
+            ApplicationEntity applicationEntity;
+            VersionEntity versionEntity;
+            try {
+
+                String versionName = getVersionName(event, instance);
+                 versionEntity = new VersionEntity(versionName);
+
+                String applicationName = getApplicationName(event, instance);
+                 applicationEntity = new ApplicationEntity(applicationName);
+            }
+            catch (AmazonServiceException e) {
+                LOG.warn("Lifecycle: {}", e.getMessage());
```

Comment:
would be nice to have a more speaking error message: "Could not get version/application for lifecycle event."
I personally prefer e.toString() over e.getMessage() because it contains the exception name as well. But you could also use the slf4j api here: LOG.warn(String msg, Throwable t) which prints the whole stack trace. This might be very useful.